File: 1446_consecCharsMaxPow.cpp
// 1446. Consecutive Characters

class Solution {
public:
    int maxPower(string s) {
        
        vector<int> v;
        
        for(int i=0;i<s.length();i++){
            
            if(i==0 || s[i]!=s[i-1])
                v.push_back(1);
            else{
                v.back() = v.back()+1;
            }
        }
        
        int max_val = 0;
        
        for(int i=0;i<v.size();i++){
            
            if(v.at(i)>max_val)
                max_val = v.at(i);
        }
        
        return max_val; 
    }
};

File: 1446_consecCharsMaxPow.py
# 1446. Consecutive Characters

class Solution:
    def maxPower(self, s: str) -> int:
        
        hash_map = []
        
        for i in range(len(s)):
            
            if i==0 or s[i]!=s[i-1]:
                hash_map.append([s[i], 1])
            else:
                hash_map[-1][1]+=1 
        
        max_val = 0       
        
        for i in range(len(hash_map)):
            
            if hash_map[i][1]>max_val:
                max_val=hash_map[i][1]
        
        return max_val
